src/components/text-editor/utils/html-converter.ts

```diff
@@ -1,14 +1,27 @@
 import { ContentTypeConverter } from './content-type-converter';
 import { EditorView } from 'prosemirror-view';
+import { markdownToHTML } from '../../markdown/markdown-parser';
+import { CustomElementDefinition } from 'src/interface';
 
 /**
  * @private
  */
 export class HTMLConverter implements ContentTypeConverter {
+    private customNodes: CustomElementDefinition[];
+
+    constructor(plugins: CustomElementDefinition[]) {
+        this.customNodes = plugins;
+    }
+
     public parseAsHTML = (text: string): Promise<string> => {
-        return new Promise((resolve) => {
-            resolve(text);
-        });
+        const whitelist: CustomElementDefinition[] = this.customNodes.map(
+            (nodeConfig: CustomElementDefinition) => ({
+                tagName: nodeConfig.tagName,
+                attributes: nodeConfig.attributes,
+            }),
+        );
+
+        return markdownToHTML(text, { whitelist: whitelist });
```

Comment:
Why do we need the map here? Maybe I'm missing something? :)
```suggestion
        return markdownToHTML(text, { whitelist: this.customNodes });
```

Reply:
Good catch! I just took it from `markdownConverter`, assuming it did something. But it doesn't seem to be doing anything at all 😅